tests/sles4sap/qesapdeployment/test_console_redirection.pm

```diff
@@ -0,0 +1,49 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Test for qe-sap-deployment
+# Maintainer: QE-SAP <qe-sap@suse.de>, Michele Pagot <michele.pagot@suse.com>
+
+use strict;
+use warnings;
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use sles4sap::qesap::qesapdeployment;
+use sles4sap::console_redirection;
+use YAML::PP;
+
+sub run {
+    my ($self) = @_;
+
+    my $current_id = script_output('cat /etc/machine-id', quiet => 1);
+    my $remote_id;
+
+    my $ypp = YAML::PP->new;
+    my $raw_file = script_output('cat ' . qesap_get_inventory(provider => get_required_var('PUBLIC_CLOUD_PROVIDER')));
+    my $inventory_data = $ypp->load_string($raw_file)->{all}{children};
+
+    for my $hostname (keys %{$inventory_data->{hana}{hosts}}) {
+        next unless $hostname =~ /vmhana/;
+        connect_target_to_serial(
+            destination_ip => $inventory_data->{hana}{hosts}{$hostname}{ansible_host},
+            ssh_user => 'cloudadmin',
+            switch_root => 1);
+        check_serial_redirection();
+
+        script_run('hostname');
+        $remote_id = script_output('cat /etc/machine-id', quiet => 1);
+
+        disconnect_target_from_serial();
+    }
+}
+
+sub post_fail_hook {
+    my ($self) = shift;
+    disconnect_target_from_serial() if check_serial_redirection();
```

Comment:
You don't need the if statement here as it is already included in `disconnect_target_from_serial`

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/da44c13e2e8740550957a1786184965f638fb42b/lib/sles4sap/console_redirection.pm#L163